clients/js/src/helpers/fetch.ts

```diff
@@ -147,6 +147,46 @@ export const fetchAsset = async (
   return deriveAssetPlugins(assetV1, await fetchCollectionV1(umi, collection));
 };
 
+/**
+ * Helper function to fetch multiple assets and derive plugins from their collections if applicable.
+ *
+ * @param umi Context
+ * @param assets Array of asset addresses to fetch
+ * @param options Options, `skipDerivePlugins` plugins from collection is false by default
+ * @returns Promise of a list of `AssetV1`
+ */
+export const fetchAllAssets = async (
+  umi: Context,
+  assets: Array<PublicKey | string>,
+  options: { skipDerivePlugins?: boolean } & RpcGetAccountOptions = {}
+): Promise<AssetV1[]> => {
+  const assetV1s = await Promise.all(
+    assets.map((asset) => fetchAssetV1(umi, publicKey(asset)))
```

Comment:
i think we will want to use a getMultipleAccounts here (i think this is fetchAllAssetsV1) rather than fire off many rpc requests. There should probably be chunking too, since iirc GMA will have a max number of accounts allowed in a single rpc request

Reply:
Added a chunkSize option with default 1000.